src/main.ino

```diff
@@ -112,8 +98,8 @@ void write_data_line( )
     unsigned long timestamp = millis() - trial_start_time_;
 
     char msg[100];
-    sprintf(msg, "%lu,%d,%d,%d,%d,%d,%d,%s,%ld"
-            , timestamp, trial_count_, puff, tone, led
+    sprintf(msg, "%lu,%s,%d,%d,%d,%d,%d,%d,%s,%ld"
```

Comment:
Note the format of data which arduino sends and we integrate into first row of frame. I've added current protocol code as the second field. Rest remains the same. @ananthamurthy You may have to change your matlab code. Also note that last two values deals with rotary encoder. Second last is count of steps (mostly useless) but the last one is rotation speed (`+` means counterclock-wise, `-` means clockwise | I may be wrong here).